compiler/rustc_hir_analysis/src/check/compare_impl_item.rs

```diff
@@ -345,7 +345,7 @@ fn compare_method_predicate_entailment<'tcx>(
 
         let emitted = report_trait_method_mismatch(
             infcx,
-            cause,
+            cause.clone(),
```

Comment:
Why this clone? This is an early return and `cause` is not used again before the early return